packages/pigeon/lib/cpp_generator.dart

```diff
@@ -871,11 +871,9 @@ class CppSourceGenerator extends StructuredGenerator<CppOptions> {
           scope: api.name,
           returnType: _voidType,
           parameters: parameters, body: () {
-        const String channel = 'channel';
         indent.writeln(
             'const std::string channel_name = "${makeChannelName(api, func, dartPackageName)}";');
-        indent.writeln(
-            'auto channel = std::make_unique<BasicMessageChannel<>>(binary_messenger_, '
+        indent.writeln('BasicMessageChannel<> channel(binary_messenger_, '
```

Comment:
These changes should ideally have a Dart unit test; that's generally how we handle changes that affect generation in some desirable way, but don't change observed behavior.

Reply:
Gotcha, I added this but let me know if this isn't what you had in mind: https://github.com/flutter/packages/pull/6196/files#diff-e8ad868911157f965b470ee46bff77cea5651550151081a6114d6a445ac2cc2fR2118